tough/src/editor/mod.rs

```diff
@@ -193,12 +232,24 @@ impl RepositoryEditor {
         Ok(self)
     }
 
+    /// Returns a mutable reference to the targets editor if it exists, or an error if it doesn't
+    fn targets_editor_mut(&mut self) -> Result<&mut TargetsEditor<'a, T>> {
+        self.targets_editor
+            .as_mut()
```

Comment:
Is this necessary? Don't we own `targets_editor`?

Reply:
I think we need that to convert `&mut Option<TargetsEditor>` to `Option<&mut TargetsEditor>` so that we can unwrap without moving.